amazon-kinesis-client-multilang/src/main/java/software/amazon/kinesis/multilang/config/KinesisClientLibConfigurator.java

```diff
@@ -62,14 +64,34 @@ public KinesisClientLibConfigurator() {
     public MultiLangDaemonConfiguration getConfiguration(Properties properties) {
         properties.entrySet().forEach(e -> {
             try {
-                utilsBean.setProperty(configuration, (String) e.getKey(), e.getValue());
+                Object value = e.getValue();
+                if(value instanceof String){
+                    value = ((String) value).trim();
+                }
+                utilsBean.setProperty(configuration, (String) e.getKey(), value);
             } catch (IllegalAccessException | InvocationTargetException ex) {
                 throw new RuntimeException(ex);
             }
         });
 
         Validate.notBlank(configuration.getApplicationName(), "Application name is required");
-        Validate.notBlank(configuration.getStreamName(), "Stream name is required");
+
+        try {
+            Validate.notBlank(configuration.getStreamArn(), "");
+            Arn streamArnObj = Arn.fromString(configuration.getStreamArn());
+
+            //Parse out the stream Name from the Arn (and/or override existing value for Stream Name)
+            String streamNameFromArn = streamArnObj.getResource().getResource();
+            configuration.setStreamName(streamNameFromArn);
+
+            //Parse out the region from the Arn and set (and/or override existing value for region)
+            String regionName = streamArnObj.getRegion();
+            Region regionObj = Region.of(regionName);
+            configuration.setRegionName(regionObj);
+        }catch (Exception e) {
+            Validate.notBlank(configuration.getStreamName(), "Stream name or Stream Arn is required. (Stream Arn takes precedence if both are passed in)");
```

Comment:
Is it correct that if Stream Arn is passed in, then Stream Name is skipped? If Stream Arn throws an exception, then Stream Name is checked?

Reply:
Correct, if both are passed in Stream Arn will take precedence.